app/vmselect/promql/rollup.go

```diff
@@ -2466,14 +2467,15 @@ func getIntNumber(arg any, argNum int) (int, error) {
 	return n, nil
 }
 
+// getString expects result from a string expression, which contains a single timeseries with only NaN values.
 func getString(tss []*timeseries, argNum int) (string, error) {
 	if len(tss) != 1 {
-		return "", fmt.Errorf(`arg #%d must contain a single timeseries; got %d timeseries`, argNum+1, len(tss))
+		return "", fmt.Errorf(`arg #%d must be a string`, argNum+1)
```

Comment:
this error message is confusing. It checks for the number of arguments passed but message complains it is not a string.
For example, `count_values_over_time(union("foo"), series_selector[5m])` will work and don't throw error message. But `count_values_over_time(union("foo", "bar"), series_selector[5m])` will throw an error message about arg1 not being a string. This is confusing.

Reply:
I think the error message is correct and friendly for most of the users: the argument must be a string(under the hood, a string timeseries is used), regardless of how it’s generated.
>count_values_over_time(union("foo"), series_selector[5m])

This example works because the query result of `union("foo")` equals `"foo"`: a string timeseries using "foo" as the MetricGroup with only NaN values. In prometheus, it returns result with `resultType: "string"`, but in MetricsQL, the result is eventually removed as empty series in https://github.com/VictoriaMetrics/VictoriaMetrics/blob/25cd5637bc5573387597a58f80264b83d21d89cc/app/vmselect/promql/exec.go#L193

With `union("foo", "bar")`, the result contains two strings timeseries, so it fails the `getString()` check.